hazelcast/src/main/java/com/hazelcast/spi/impl/operationservice/impl/OperationRunnerImpl.java

```diff
@@ -311,7 +312,9 @@ public void run(Packet packet) throws Exception {
     }
 
     private boolean ensureValidMember(Operation op) {
-        if (isJoinOperation(op) || node.clusterService.getMember(op.getCallerAddress()) != null) {
+        if (isJoinOperation(op)
+                || isWanReplicationOperation(op)
+                || node.clusterService.getMember(op.getCallerAddress()) != null) {
```

Comment:
this can be short-circuited as well - as majority of operations have a valid caller set.